src/Exceptions/InvalidCurrencyException.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions;
+
+class InvalidCurrencyException extends \Exception
+{
```

Comment:
Add the custom constructor here to set the message in here rather than in the calling method.